protocol/synthetix/test/integration/storage/CollateralConfiguration.test.ts

```diff
@@ -0,0 +1,167 @@
+import assertRevert from '@synthetixio/core-utils/utils/assertions/assert-revert';
+import assertBn from '@synthetixio/core-utils/utils/assertions/assert-bignumber';
+import { bootstrap } from '../bootstrap';
+import { ethers } from 'ethers';
+import hre from 'hardhat';
+
+describe('CollateralConfiguration', function () {
+  const { systems, signers } = bootstrap();
+
+  let owner: ethers.Signer;
+
+  let fakeCollateral: ethers.Contract;
+
+  const ONE = ethers.utils.parseEther('1');
+
+  const initFakeCollateralConfig = async (decimals: number) => {
+    [owner] = signers();
+
+    const CollateralMock = await hre.ethers.getContractFactory('CollateralMock');
+    fakeCollateral = await CollateralMock.deploy();
+    await fakeCollateral.deployed();
+    await fakeCollateral.initialize('token', 'TOKEN', decimals);
+    await (
+      await systems()
+        .Core.connect(owner)
+        .configureCollateral({
+          tokenAddress: fakeCollateral.address,
+          oracleNodeId: ethers.utils.formatBytes32String(''),
+          issuanceRatioD18: '5000000000000000000',
+          liquidationRatioD18: '1500000000000000000',
+          liquidationRewardD18: '20000000000000000000',
+          minDelegationD18: '20000000000000000000',
+          depositingEnabled: true,
+        })
+    ).wait();
+  };
+
+  describe('convertTokenToSystemAmount()', async () => {
+    it('scales token with 0 decimals to system amount', async () => {
+      const DECIMALS = 0;
+      await initFakeCollateralConfig(DECIMALS);
```

Comment:
for all the tests that use `initFakeCollateralConfig` would be best to enclose in a `describe` block that has a `before`

I know this is probably normal for foundry but in our hardhat project tests are highly heiarchal.